lib/python/picongpu/input/parameters.py

```diff
@@ -175,8 +184,10 @@ def as_dict(self):
         """
 
         members = super(UiParameter, self).as_dict()
-        del members["formatter"]
-        del members["dtype"]
+        members["label"] = self.label
+        members["formatter"] = str(inspect.getsourcelines(self.formatter)[
```

Comment:
did you try this with multi-line formatters?

Reply:
No, I haven't done so yet...

Reply:
we should add that to avoid bad side effects if users do not pass a one-line lambda

Reply:
Then maybe we could use something like this:
```
members["formatter"] = ' '.join(inspect.getsourcelines(self.formatter)[0]).strip()
```

Reply:
but how to de-serialize it again to clean multi-lines?